delphi/translators/for2py/pyTranslate.py

```diff
@@ -1142,6 +1143,92 @@ def get_python_source(self):
 
         return "".join(self.pyStrings)
 
+    """
+        This function checks the type of a variable and returns the appropriate python syntax type name
+    """
+    def get_type(self, node):
+        variable_type = node["type"].lower()
+        if variable_type in TYPE_MAP: 
+            return TYPE_MAP[variable_type]
+        else:
+            if node["is_derived_type"] == "true":
+                return variable_type
+            else:
+                assert False, f"Unrecognized variable type: {variable_type}"
+
+    """
+        This function will construct the range string in 'loBound, Upbound' format and return to the called function
+    """
+    def get_range(self, node):
+        loBound = "0"
+        upBound = "0" 
+
+        low = node["low"]
+        up = node["high"]
+        # Get lower bound value
+        if low[0]["tag"] == "literal":
+            loBound = self.proc_literal(low[0])
+        elif low[0]["tag"] == "op":
+            loBound = self.proc_op(low[0])
+        else:
+            assert False, f"Unrecognized tag in upper bound: {low[0]['tag']}"
+
+        # Get upper bound value
```

Comment:
The code to get the upper and lower bounds is duplicated.  I would prefer to have a single piece of code (a separate method) that gets the bound, and which is called with `low` and `up` (or `low[0]` and `up[0]`) as arguments.

More generally, I would prefer that duplicated code be avoided unless there are strong compelling reasons for the duplication.